File: base/src/main/java/com/zhushenwudi/base/wifi/IWifi.java
package com.zhushenwudi.base.wifi;

public interface IWifi {

    String name();

    boolean isEncrypt();

    boolean isSaved();

    boolean isConnected();

    String encryption();

    int level();

    String description();

    String ip();

    String description2();

    void state(String state);

    void setConnected(Boolean flag);

    @Deprecated
    String SSID();

    @Deprecated
    String capabilities();

    @Deprecated
    IWifi merge(IWifi merge);

    String state();
}

File: base/src/main/java/com/zhushenwudi/base/wifi/IWifi.kt
package com.zhushenwudi.base.wifi

interface IWifi {
    fun name(): String?
    var isEncrypt: Boolean
    var isSaved: Boolean
    var isConnected: Boolean
    fun encryption(): String?
    fun level(): Int
    fun description(): String?
    fun ip(): String?
    fun description2(): String?
    fun state(state: String?)

    fun SSID(): String?

    fun capabilities(): String?

    fun merge(merge: IWifi?): IWifi?
    fun state(): String?
}